torch/csrc/jit/script/module.h

```diff
@@ -125,17 +126,117 @@ struct TORCH_API Method {
   }
 
  private:
-  // Methods are uniqued onwed by a single module. This raw pointer allows
-  // looking up the module.
-  ModulePtr owner_;
+  // Methods are uniqued onwed by a single object. This raw pointer allows
+  // looking up the object.
+  ObjectPtr owner_;
 
   // Underlying unbound function
   // This is the _lowered_ function and is different than the
   // first-class function in class_compilation_unit()
   Function* function_;
 };
 
-struct TORCH_API Module {
+struct TORCH_API Object {
```

Comment:
Object should go in its own header and compilation unit. It will grow over time, and it is confusing for it to be in module.h